src/crimson/os/seastore/cache.h

```diff
@@ -226,18 +226,26 @@ class Cache {
   get_extent_if_cached_ret get_extent_if_cached(
     Transaction &t,
     paddr_t offset) {
-    return seastar::do_with(
-      CachedExtentRef(),
-      [this, &t, offset](auto &ret) {
-	auto status = query_cache_for_extent(t, offset, &ret);
-	auto wait = seastar::now();
-	if (status == Transaction::get_extent_ret::PRESENT) {
-	  wait = ret->wait_io();
-	}
-	return trans_intr::make_interruptible(
-	  wait.then([ret] { return std::move(ret); })
-	);
-      });
+    CachedExtentRef ret;
+    auto result = t.get_extent(offset, &ret);
+    if (result != Transaction::get_extent_ret::ABSENT) {
+      // including get_extent_ret::RETIRED
```

Comment:
I just stick to the original logic here, but is returning retired extent correct? @athanatos

Reply:
It's not really ok to request an extent the transaction has already invalidated, so that's fine.